src/Features/LanguageServer/Protocol/Handler/Completion/CompletionResolveHandler.cs

```diff
@@ -98,12 +102,20 @@ private static bool MatchesLSPCompletionItem(LSP.CompletionItem lspCompletionIte
                 return false;
             }
 
-            if (!lspCompletionItem.Label.EndsWith(completionItem.DisplayTextSuffix, StringComparison.Ordinal))
+            // The prefix matches, consume the matching prefix from the lsp completion item label.
+            var originalDisplayText = lspCompletionItem.Label.Substring(completionItem.DisplayTextPrefix.Length, lspCompletionItem.Label.Length);
+
+            if (!originalDisplayText.EndsWith(completionItem.DisplayTextSuffix, StringComparison.Ordinal))
             {
                 return false;
             }
 
-            if (string.Compare(lspCompletionItem.Label, completionItem.DisplayTextPrefix.Length, completionItem.DisplayText, 0, completionItem.DisplayText.Length, StringComparison.Ordinal) != 0)
+            // The suffix matches, consume the matching suffix from the lsp completion item label.
+            originalDisplayText = originalDisplayText.Substring(0, originalDisplayText.Length - completionItem.DisplayTextSuffix.Length);
```

Comment:
I'd just throw this above with the prefix handling. Otherwise the `originalDisplayText` isn't so "original" until this is done 😄

Reply:
I like how this method reads as-is, so I'll adjust the naming instead